packages/drivers/routerlicious-driver/src/policies.ts

```diff
@@ -9,4 +9,14 @@ export interface IRouterliciousDriverPolicies {
      * Default: true
      */
     enablePrefetch: boolean;
+    /**
+     * Rate limit concurrent storage requests.
+     * Default: 100
+     */
+    maxConcurrentStorageRequests: number;
+    /**
+     * Rate limit concurrent orderer requests.
+     * Default: 100
+     */
+    maxConcurrentOrdererRequests: number;
```

Comment:
Sorry if it's dump question, but orderer is equivalent to socket, no? What is orderer requests?
Also, should it be a single limit for factory? Otherwise each sub-bucket needs to be at most half of what browser supports, not allowing you to achieve max throughput in most scenarios (and if sum of these two limits is a bit higher than browser limit, then very small number of users will experience bugs and it would be very hard to diagnose them).

Reply:
I am a bit confused as well. Does ordererrequests refer to fetch deltas call?

Reply:
Not a dumb question. Our Orderer is also where we fetch Deltas from, so there are HTTP requests going to Alfred. Thinking about it though, those are never in parallel, so I can remove this and the RateLimiter for OrdererRestWrapper (or just set it to 1)

Reply:
> Otherwise each sub-bucket needs to be at most half of what browser supports, not allowing you to achieve max throughput in most scenarios (and if sum of these two limits is a bit higher than browser limit, then very small number of users will experience bugs and it would be very hard to diagnose them).

To answer this part of the question, since the Orderer is separate from Storage, they are 2 different hosts, and therefore would have separate connections, which means that they would have separate "parallel requests per connection" limits. That was why I had separated the rate limiter configs at the per-"service" level, rather than the per-operation level.

Reply:
I'd guess that browser starts failing fetch request when total number of pending (sitting in browser queue) requests across all domains is too high. I.e. different domains used might reduce such pending queue, but I have a feeling that in grand schema of things it will not have such a bit impact. It would be great to confirm or deny that.

Reply:
Looks like HTTP/2 in R11s is (currently) configured server-side, via Nginx defaults, such that the client (browser) can send [1000 max total requests per connection](http://nginx.org/en/docs/http/ngx_http_v2_module.html#http2_max_requests), and [128 max concurrent requests (streams) per connection](http://nginx.org/en/docs/http/ngx_http_v2_module.html#http2_max_concurrent_streams). According to [RFC7540 - Hypertext Transfer Protocol Version 2 (HTTP/2) - Section 6.5.2](https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2), the max concurrent streams value should be no lower than 100.

This seems to indicate that it is on a per-connection basis, not a per-browser-queue limit. For now, I think setting Rate Limit to 100 is a good, safe starting point, but I can do some manual testing. So far, the only browser-imposed request limits I've found are connections-per-domain (6-8 for HTTP/1.1, 1 for HTTP/2) and concurrent streams (100 for most browsers, unclear if this limits requests or server pushes, but it appears to only affect max server pushes).

Reply:
How many connections browser makes per domain? I think most use 6 for HTTP/1.1, not sure about HTTP/2

Reply:
1 for HTTP/2 ([One Connection Per Origin](https://developers.google.com/web/fundamentals/performance/http2#one_connection_per_origin))